src/Drupal/Commands/core/WatchdogCommands.php

```diff
@@ -62,9 +64,30 @@ public function show($substring = '', $options = ['format' => 'table', 'count' =
             $row = $this->formatResult($result, $options['extended']);
             $table[$row->wid] = (array)$row;
         }
-        if (empty($table)) {
+        if (empty($table) && !$tail) {
             $this->logger()->notice(dt('No log messages available.'));
         } else {
+            if ($tail) {
+                // Query for new watchdog messages every 2 seconds.
+                while (true) {
+                    $rsc = Database::getConnection()->select('watchdog', 'w')
+                        ->fields('w')
+                        ->condition('wid', $last_wid, '>')
+                        ->orderBy('wid', 'ASC')
+                        ->execute();
```

Comment:
This query doesn't respect the $where clause.  I'm not sure if it needs to, but I figured I'd point it out.